server/routes/Categories.js

```diff
@@ -0,0 +1,44 @@
+const express = require('express');
+const Category = require('../models/Category.model');
+const { TODOItem } = require("../models/TODOItem.model");
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+  try {
+    const outputList = await Category.find().sort({ category: 1 });
+    res.send(outputList);
+  } catch (error) {
+    console.error('Error fetching categories:', error);
+    res.status(500).send('An error occurred while fetching categories: ' + error.message);
+  }
+});
+
+router.delete('/:categoryID', async (req, res) => {
+  try {
+    const items = await TODOItem.find({
+      category: req.params.categoryID,
+    });
+
+    // Do not delete the category if there are TODOItems that belong to it
+    if (items.length) {
+      return res
+        .status(409)
+        .send(
+          'The category is still used by other TODO Items. So, the category cannot be deleted'
```

Comment:
There is no user feedback for a failed deletion of a category. The user just sees the component reloading while the category and its associated TODOs remain.

Reply:
This part for some reason is a bit tricky to implement. I will look into it again

Reply:
![image](https://github.com/awctw/study-dash/assets/56493743/311432fc-8dcb-44bf-8a6b-17f8918a6cba)

Maybe add a period to the end and make the message more concise? Something like, "This category is still in use and so it cannot be deleted." or, "Categories still in use cannot be deleted."

Reply:
Nice suggestion. I will make this fix next